docs/machine-learning/tutorials/sales-anomaly-detection.md

```diff
@@ -0,0 +1,372 @@
+
+Create a display header line using the following <xref:System.Console.WriteLine?displayProperty=nameWithType> code:
+
+[!code-csharp[DisplayHeader1](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayHeader1)]
+
+You'll display the following information in your spike detection results:
+
+* `Alert` indicates a spike alert for a given data point.
+
+* `Score` is the `ProductSales` value for a given data point in the dataset.
+
+* `P-Value` The "P" stands for probability. This indicates how likely this data point is an anomaly. 
+
+Use the following code to iterate through the `predictions` `IEnumerable` and display the results:
+
+[!code-csharp[DisplayResults1](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#DisplayResults1)]
+
+## Spike detection results
+
+Your results should be similar to the following. During processing, messages are displayed. You may see warnings, or processing messages. These have been removed from the following results for clarity.
+
+```console
+Detect temporary changes in pattern
+=============== Training the model ===============
+=============== End of training process ===============
+Alert   Score   P-Value
+0       271.00  0.50
+0       150.90  0.00
+0       188.10  0.41
+0       124.30  0.13
+0       185.30  0.47
+0       173.50  0.47
+0       236.80  0.19
+0       229.50  0.27
+0       197.80  0.48
+0       127.90  0.13
+1       341.50  0.00 <-- Spike detected
+0       190.90  0.48
+0       199.30  0.48
+0       154.50  0.24
+0       215.10  0.42
+0       278.30  0.19
+0       196.40  0.43
+0       292.00  0.17
+0       231.00  0.45
+0       308.60  0.18
+0       294.90  0.19
+1       426.60  0.00 <-- Spike detected
+0       269.50  0.47
+0       347.30  0.21
+0       344.70  0.27
+0       445.40  0.06
+0       320.90  0.49
+0       444.30  0.12
+0       406.30  0.29
+0       442.40  0.21
+1       580.50  0.00 <-- Spike detected
+0       412.60  0.45
+1       687.00  0.01 <-- Spike detected
+0       480.30  0.40
+0       586.30  0.20
+0       651.90  0.14
+```
+
+## Change point detection
+
+`Change points` are persistent changes in a time series event stream distribution of values, like level changes and trends. These persistent changes last much longer than `spikes` and could indicate catastrophic event(s). `Change points` are not usually visible to the naked eye, but can be detected in your data using approaches such as in the following method.  The following image is an example of a change point detection:
+
+![ChangePointDetection](./media/sales-anomaly-detection/ChangePointDetection.png)
+
+### Create the DetectChangepoint() method
+
+Add the following call to the `DetectChangepoint()`method as the next line of code in the `Main()` method:
+
+[!code-csharp[CallDetectChangepoint](~/samples/machine-learning/tutorials/ProductSalesAnomalyDetection/Program.cs#CallDetectChangepoint]
```

Comment:
Check this code snippet. Might be broken.

Reply:
Yup. Missing an ending parenthesis. Good catch!